webargs/pyramidparser.py

```diff
@@ -112,8 +115,10 @@ def wrapper(obj, *args, **kwargs):
                     request = req or obj.request
                 except AttributeError:  # first arg is request
                     request = obj
-                parsed_args = self.parse(schema, req=request, locations=locations,
-                                         validate=validate, force_all=as_kwargs)
+
+                parsed_args = self.parse(schema or argmap(request), req=request,
```

Comment:
I considered this however but felt it wasn't overly useful since if I'm calling ```parse``` directly then I'm already going to have access to the request and therefore I can construct the schema precisely as I want. Is there some use case I'm missing or is it purely about pushing the logic in ```parse``` to save the duplication around the ```schema or argmap(request)``` style logic which presently occurs in a few places?